imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java

```diff
@@ -981,23 +982,25 @@ private int dpToPixel(int dp) {
         return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, getResources().getDisplayMetrics());
     }
 
-    public void saveState() {
+    public ViewState saveState() {
         savedAspectRatioWidth = mAspectRatioWidth;
         savedAspectRatioHeight = mAspectRatioHeight;
         mSuppMatrix.getValues(suppMatrixValues);
+
+        return new ViewState(getImageViewMatrix(), mAspectRatioWidth, mAspectRatioHeight, suppMatrixValues);
     }
 
-    public void restoreState() {
+    public void restoreState(ViewState viewState) {
         mBitmapChanged = true;
         mRestoreRequest = true;
-        mAspectRatioWidth = savedAspectRatioWidth;
-        mAspectRatioHeight = savedAspectRatioHeight;
```

Comment:
@rnitame 
I think `savedAspectRatioWidth`, `savedAspectRatioHeight` and `suppMatrixValues` fields don't need anymore. isn't it?

Reply:
`savedAspectRatioWidth` and `savedAspectRatioHeight` fields are unnecessary 🙆 I will delete.
But I could not restore without `suppMatrixValues` field.

Reply:
You deleted `aspectRatioWidth` and `aspectRatioHeight` in ViewState. isn't that right?
I mentioned about `savedAspectRatioWidth` and `savedAspectRatioHeight`

Reply:
> You deleted aspectRatioWidth and aspectRatioHeight in ViewState. isn't that right?

Right.

> I mentioned about savedAspectRatioWidth and savedAspectRatioHeight

It means this 2 lines? 

https://github.com/naver/android-imagecropview/blob/develop/imagecropview/src/main/java/com/naver/android/helloyako/imagecrop/view/ImageCropView.java#L132-L133

Reply:
updated https://github.com/naver/android-imagecropview/compare/04a5dfe41e3f88447146936130c5e473f9c2fc30..3f7ad5a525085c45f158d399017d082235e1bc39